### PR TITLE
set plugins in player config object from uiConfig data

### PR DESCRIPTION
**Pull request by tanyaLibatter**

### CODE REVIEW

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 1]
```diff
@@ -179,6 +179,7 @@ private function getAutoEmbedCode($targetId = null)
 
 		$config["partnerId"] = $this->partnerId;		
 		$config["uiConfId"] = $this->uiconfId;
+		$config["plugins"] = $this->playerPlugins;
```
Comment (OrenMe) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: This needs to be added also to dynamic embed

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 2]
```diff
@@ -272,6 +273,9 @@ private function initMembers()
 		$uiConf = uiConfPeer::retrieveByPK($this->uiconfId);
 		if (!$uiConf)
 			KExternalErrors::dieError(KExternalErrors::UI_CONF_NOT_FOUND);
+
+		$playerConfig = json_decode($uiConf->getConfig(), true);
```
Comment (OrenMe) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: the uiconf config can be any arbitrary json config, not only plugins config.
Basically you need to assume that getConfig will return the entire config object of the player.

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php (hunk 1 above)
Comment (tanyaLibatter) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: done

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php (hunk 2 above)
Comment (tanyaLibatter) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: done

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 3]
```diff
@@ -177,8 +188,11 @@ private function getAutoEmbedCode($targetId = null)
 			$config[$key] = json_decode($val);
 		}
 
+
 		$config["partnerId"] = $this->partnerId;		
 		$config["uiConfId"] = $this->uiconfId;
+		$config["uiConf"] = $this->playerConfig;
```
Comment (yossipapi) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: playerConfig is missing in the class params definition.

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 4]
```diff
@@ -272,6 +286,8 @@ private function initMembers()
 		$uiConf = uiConfPeer::retrieveByPK($this->uiconfId);
 		if (!$uiConf)
 			KExternalErrors::dieError(KExternalErrors::UI_CONF_NOT_FOUND);
+
+		$this->playerConfig = json_decode($uiConf->getConfig(), true);
```
Comment (yossipapi) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: playerConfig is missing in the class params definition.

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php (hunk 3 above)
Comment (OrenMe) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: @tanyaLibatter there's no uiConf in the player code, you need to pass the config as is.
The config option would be same format as player expects it to be, e.g. 
```json
{
   "playback": {...},
   "plugins": {
      "pluginA": {...},
      "pluginB": {...}
   }
}
```

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php [hunk 5]
```diff
@@ -108,6 +108,17 @@ private function formatBundleContent($bundleContent)
 		{
 			$bundleContent = $this->getIfarmEmbedCode($bundleContent);
 		}
+		else
+		{
+			$config["uiConf"] = $this->playerConfig;
+			$config = json_encode($config);	
+
+			$kalturaPlayerConfig = "
+			var kalturaPlayerConfig = $config;
```
Comment (OrenMe) [inline alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: how do you expect the player to get the default config? we will need to add some relevant code in player that takes this into consideration when it's loaded.
Currently this kalturaPlayerConfig is just lying there and the app needs to refer to it when calling `KalturaPlayer.setup(..., config);`

Comment (OrenMe) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: Also, need to make this config take into consideration a situation where multiple players are on page and we don't want them to run over another

File: alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php (hunk 3 above)
Comment (OrenMe) [reply alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php]: Also, need to handle situation where config is passed from client side and the we need to merge accordingly.

**Pull request merged**